lib/newProfile.ml

```diff
@@ -78,7 +78,7 @@ let global_start = gettime()
 let global_start_stat = Gc.quick_stat()
 
 let duration ~time name ph ?args ?(last=",") () =
-  f "%a%s\n" MiniJson.pr (MiniJson.duration ~name ~ph ~ts:(prtime time) ?args ()) last
+  f "%a%s\n%!" MiniJson.pr (MiniJson.duration ~name ~ph ~ts:(prtime time) ?args ()) last
```

Comment:
It is not obvious to me if this function is called often or only once, and if this flush is to string or to disk. But just in case, keep in mind that a flush to disk is a very costly system call, especially if the target partition happens to be encrypted. (That is why flushes were removed from the OCaml compiler a few years ago.)

Reply:
@silene I understand @SkySkimmer did this on purpose, so he could see the progress on disk.

The current implementation indeed allows the client to select the formatter to use, `coqc` currently writes to disk indeed.

Reply:
I agree that this could be costly, but I'm unsure how to balance the desire for eager output with the desire of less FS trashing.

Reply:
It's certainly not called only once but I don't think it's called very often either.
Certainly the bench doesn't seem to have noticed the change.

Reply:
> Certainly the bench doesn't seem to have noticed the change.

I doubt the bench machine is using an encrypted partition.

For what is worth, the OCaml compiler was made 5x faster by removing flushes. (I am not exaggerating when I write that flushes are very expensive in some settings.)

Reply:
@silene I agree with you on how expensive the flushes are, on the other hand @SkySkimmer would like to have the data written to disk as it happens as a tracing tool.

What would you propose? Indeed there is a latency / throughtput tradeoff here.

What I would like is that the API is a bit richer so the client controls the flush, but I volunteer to implement that once I hook this tracing module to coq-lsp.

Reply:
Personally I don't mind if we don't have the flush, it was a suggestion from @ejgallego in the previous PR